curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/63ff75fd4b20c31c23f8802e.md

```diff
@@ -0,0 +1,144 @@
+---
+id: 63ff75fd4b20c31c23f8802e
+title: Step 7
+challengeType: 0
+dashedName: step-7
+---
+
+# --description--
+
+Now that you have the data you want, what's next is to populate it on the UI. But the URL returns an array of 26 authors, you won't want to populate all of them on the UI at the same time for better performance.
+
+What you should do instead is to populate 8 authors at a time. And then add a functionality to populate 8 more on each demand until there's no data to display anymore.
+
+Create 2 variables named `startingIndex` and `endingIndex`, and assign them the number values `0` and `8` respectively. Also, create a `authorDataArr` variable and set it to an empty array.
+
+**Note**: You will use the `startingIndex` and `endingIndex` variables to paginate the authors, so you can get only 8 at first. You will also put the data (an array) into the empty array of the `authorData` soon.
```

Comment:
```suggestion
Now that you have the data you want, you can use it to populate the UI. But the fetched data contains an array of 26 authors, and if you add them all to the page at the same time, it could lead to poor performance.

Instead, you should add 8 authors at a time, and have a button to add 8 more until there's no data to display.

Use `let` to create 2 variables named `startingIndex` and `endingIndex`, and assign them the number values `0` and `8`, respectively. Also, create an `authorDataArr` variable with `let` and set it to an empty array.
```